index.js

```diff
@@ -60,6 +60,15 @@ module.exports = class handlePeers extends EventEmitter {
   }
 
   _onConnection (peer) {
+    if (this.node.peerInfo.id._idB58String === peer.id._idB58String) {
+      console.log("node id match - returning early in _onConnection")
+      if (node.peerBook.has(peer)) {
+        console.log("I have myself.")
+      }
+      this.node.peerBook.remove(peer)
+      return
+    }
+
```

Comment:
Is this second check needed? We shouldn't connect to our self if you filter below.